src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int RADIUS = 10;
+    private final Random random = new Random();
+    private final double parameter = random.nextDouble();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Figure circle = new Circle(parameter, colorSupplier.getRandomColor());
+    private final Figure square = new Square(parameter, colorSupplier.getRandomColor());
+    private final Figure rectangle = new Rectangle(parameter, parameter,
+            colorSupplier.getRandomColor());
+    private final Figure rightTriangle = new RightTriangle(parameter, parameter,
+            colorSupplier.getRandomColor());
+    private final Figure isoscelesTrapezoid = new IsoscelesTrapezoid(parameter,
+            parameter, parameter, colorSupplier.getRandomColor());
+    private final Figure[] figures = {circle, square, rectangle, rightTriangle, isoscelesTrapezoid};
```

Comment:
you create 5 figures just to pick one of them. And every time you will choose figure from the same circle, square, ... when you call getRandomFigure() several times.  Lets make this variables locas and use switch-case construction. We can create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability